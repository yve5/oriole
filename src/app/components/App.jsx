import React from 'react';
import Card from './Card';
import Header from './Header';
import { cvssConfig } from '../../cvss4/references/cvssConfig';

const App = () => (
  <>
    <Header />

    <div className="container">
      <div className="row">
        <div className="col-lg-8 offset-lg-2">
          {Object.entries(cvssConfig).map(([metricType, metricData]) => (
            <Card
              metricType={metricType}
              metricData={metricData}
              key={metricType}
            />
          ))}
        </div>
      </div>
    </div>
  </>
);

export default App;
