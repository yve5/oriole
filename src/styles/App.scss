// Custom.scss
// Option A: Include all of Bootstrap

// Include any default variable overrides here (though functions won't be available)
$primary: #ff7900;
$secondary: #000;
$success: #228722;
$warning: #fc0;
$danger: #cd3c14;
$info: #4170d8;

$btn-border-radius: none;
$btn-border-radius-sm: none;
$btn-border-radius-lg: none;
$btn-transition: none;

$link-color: $secondary;
$link-hover-color: $primary;

@import '../node_modules/bootstrap/scss/bootstrap';

// shared

@import 'shared/icon';

// Then add additional custom code here

.app-logo {
  @include media-breakpoint-down(md) {
    width: 32px;
    height: 32px;
    object-fit: cover;
    object-position: left;
  }
}

@include media-breakpoint-up(md) {
  html {
    height: 100vh;

    body {
      height: 100vh;

      #root {
        height: 100vh;

        .oriole__overflow {
          overflow: auto;
          max-height: 94.1vh;
        }
      }
    }
  }
}
