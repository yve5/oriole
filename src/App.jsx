import React from 'react';
import { HashRouter, Routes, Route } from 'react-router-dom';

import Paging from './routes/paging';
import Buttons from './routes/buttons';
import Layout from './features/layout';

import './styles.scss';

const App = () => (
  <HashRouter>
    <Routes>
      <Route element={<Layout />}>
        <Route index element={<Paging />} />
        <Route path="buttons" element={<Buttons />} />
      </Route>
    </Routes>
  </HashRouter>
);

export default App;
