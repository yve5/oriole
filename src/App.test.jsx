import React from 'react';
import { act, render } from '@testing-library/react';

import App from './App';

describe('App', () => {
  it.skip('renders learn react link', () => {
    const { getByText } = render(<App />);
    expect(getByText('Paging')).not.toBeNull();
  });

  it.skip('should match expected snapshot', () => {
    const { asFragment, rerender } = render(<App />);
    rerender(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
