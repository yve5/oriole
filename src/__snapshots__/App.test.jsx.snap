// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match expected snapshot 1`] = `
[
  <header
    className="sticky-top header-minimized"
  >
    <nav
      aria-label="Main navigation"
      className="navbar navbar-dark bg-dark navbar-expand-lg"
    >
      <div
        className="container-fluid"
      >
        <div
          className="navbar-brand me-auto me-lg-4"
        >
          <span
            className="stretched-link"
          >
            <img
              alt="Boosted - Back to Home"
              height="50"
              loading="lazy"
              src="/oriole/src/assets/logo.svg"
              width="50"
            />
          </span>
        </div>
        <button
          aria-controls="bd-navbar1 bd-navbar2"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler collapsed"
          data-bs-target=".bd-navbar"
          data-bs-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="navbar-collapse collapse me-lg-auto bd-navbar mb-0 border-0"
        >
          <ul
            className="navbar-nav"
          >
            <li
              className="nav-item"
            >
              <span
                className="nav-link"
              >
                Home
              </span>
            </li>
            <li
              className="nav-item"
            >
              <span
                aria-current="true"
                className="nav-link active"
              >
                Docs
              </span>
            </li>
            <li
              className="nav-item"
            >
              <span
                className="nav-link"
              >
                Examples
              </span>
            </li>
            <li
              className="nav-item"
            >
              <span
                className="nav-link"
              >
                Guidelines
              </span>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </header>,
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark mb-5"
  >
    <div
      className="container-fluid"
    >
      <button
        className="navbar-toggler"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="#/"
              onClick={[Function]}
            >
              Paging
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="#/buttons"
              onClick={[Function]}
            >
              Buttons
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div
      className="card"
    >
      <ul
        className="list-group list-group-flush"
      >
        <li
          className="list-group-item"
        >
          <table
            className="table"
          >
            <thead>
              <tr>
                <th
                  scope="col"
                >
                  #
                </th>
                <th
                  scope="col"
                >
                  First
                </th>
                <th
                  scope="col"
                >
                  Last
                </th>
                <th
                  scope="col"
                >
                  Handle
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <th
                  scope="row"
                >
                  1
                </th>
                <td>
                  Mark
                </td>
                <td>
                  Otto
                </td>
                <td>
                  @mdo
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  2
                </th>
                <td>
                  Jacob
                </td>
                <td>
                  Thornton
                </td>
                <td>
                  @fat
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  3
                </th>
                <td
                  colSpan="2"
                >
                  Larry the Bird
                </td>
                <td>
                  @twitter
                </td>
              </tr>
            </tbody>
          </table>
          <nav>
            <ul
              className="pagination"
            >
              <li
                className="page-item"
              >
                <button
                  className="page-link"
                  type="button"
                >
                  Previous
                </button>
              </li>
              <li
                className="page-item"
              >
                <button
                  className="page-link"
                  type="button"
                >
                  1
                </button>
              </li>
              <li
                className="page-item"
              >
                <button
                  className="page-link"
                  type="button"
                >
                  2
                </button>
              </li>
              <li
                className="page-item"
              >
                <button
                  className="page-link"
                  type="button"
                >
                  3
                </button>
              </li>
              <li
                className="page-item"
              >
                <button
                  className="page-link"
                  type="button"
                >
                  Next
                </button>
              </li>
            </ul>
          </nav>
        </li>
      </ul>
    </div>
  </div>,
]
`;
